support/helm/helm.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"strings"
+
+	"github.com/vladimirvivien/gexe"
+	"github.com/vladimirvivien/gexe/exec"
+	log "k8s.io/klog/v2"
+)
+
+type Opts struct {
+	// Name is used to indicate the name of the helm chart being processed
+	Name string
+	// Namespace is used to indicate the namespace in which the helm chart
+	// identified by Name will be processed
+	Namespace string
+	// ReleaseName is used to indicate the name of the release to be used
+	// while installing or upgrading the chart identified by Name into the
+	// Namespace
+	ReleaseName string
+	// Version is the helm chart version that should be deployed to the test
+	// infrastructure
+	Version string
+	// Chart is used to indicate the full path of the .tgz artifact of the Helm
+	// chart in case if your tests require you to be able to deploy a packaged
+	// helm chart that is available locally as the tarball
+	Chart string
+	// Mode is used to indicate the mode in which the helm operation is being
+	// performed. These are the first level commands that are processed by the
+	// helm Binary
+	Mode string
+	// Args is used to pass any additional arguments that you might want to pass
+	// for running the helm command in question
+	Args []string
```

Comment:
Field `Args` answers my question above.

Reply:
I have kept a set of options are directly invocable and other set of things as generic args so that we don't need to write a helper for each and every possible combination. I like @crandles  suggestion for wait being a direct invocable call for option. Will do that. Hopefully this generic `Args` should be enough for all other use